src/lib/storage/materialize.cpp

```diff
@@ -0,0 +1,35 @@
+#include "materialize.hpp"
+
+#include <type_traits>
+
+namespace opossum {
+
+std::shared_ptr<BaseColumn> materialize_as_value_column(const BaseColumn& base_column) {
```

Comment:
Is this a duplicate of #794? If not, how do they relate? Can we join them?

Reply:
This function could probably be used in `Materialize` yes.